src/App.jsx

```diff
@@ -1,248 +1,386 @@
+                      className="delete"
+                      onClick={() => setQuery('')}
+                    />
+                  </span>
+                )}
+              </p>
+            </div>
+
+            <div className="panel-block is-flex-wrap-wrap">
+              <a
+                href="#/"
+                data-cy="AllCategories"
+                className={cn(
+                  'button is-success mr-6',
+                  { 'is-outlined': sortCategory.length },
+                )}
+                onClick={() => setSortCategory([])}
+              >
+                All
+              </a>
+
+              {categories.map(category => (
+                <a
+                  data-cy="Category"
+                  className={cn(
+                    'button mr-2 my-1',
+                    { 'is-info': sortCategory.includes(category.id) },
+                  )}
+                  href="#/"
+                  onClick={() => {
+                    if (!sortCategory.includes(category.id)) {
+                      setSortCategory(prev => prev.concat(category.id));
+                    } else {
+                      setSortCategory(prev => prev
+                        .filter(id => id !== category.id));
+                    }
+                  }}
+                >
+                  {category.title}
+                </a>
+              ))}
+            </div>
+
+            <div className="panel-block">
+              <a
+                data-cy="ResetAllButton"
+                href="#/"
+                className="button is-link is-outlined is-fullwidth"
+                onClick={reset}
+              >
+                Reset all filters
+              </a>
+            </div>
+          </nav>
+        </div>
+
+        <div className="box table-container">
+          {products.length ? (
+            <table
+              data-cy="ProductTable"
+              className="table is-striped is-narrow is-fullwidth"
             >
-              Reset all filters
-            </a>
-          </div>
-        </nav>
-      </div>
-
-      <div className="box table-container">
-        <p data-cy="NoMatchingMessage">
-          No products matching selected criteria
-        </p>
-
-        <table
-          data-cy="ProductTable"
-          className="table is-striped is-narrow is-fullwidth"
-        >
-          <thead>
-            <tr>
-              <th>
-                <span className="is-flex is-flex-wrap-nowrap">
-                  ID
-
-                  <a href="#/">
-                    <span className="icon">
-                      <i data-cy="SortIcon" className="fas fa-sort" />
+              <thead>
+                <tr>
+                  <th>
+                    <span className="is-flex is-flex-wrap-nowrap">
+                      ID
+
+                      <a
+                        href="#/"
+                        onClick={() => {
+                          if (sort !== SORT_BY_ID) {
```

Comment:
Would be greate to add serparate function for sorting by specific order cause you have repeating logic for each column